### PR TITLE
amdgpu-pro vulkan and AMF bumps

### PR DESCRIPTION
**Pull request by AdelKS**
Hello!

I brought some changes that I'd like special attention to, since `ffmpeg` needs the headers from `amf-headers` to be built with the `amf` use flag :
- I expect that when `amf-headers` gets updated, `ffmpeg` needs to be rebuilt : so I slotted `amf-headers` and made `ffmpeg` get rebuilt accordingly. But I am not sure about that, it is usual to do that for apps that depend on headers right ?
- Also, I had wrongly put `amf-headers` in `BDEPEND` in `ffmpeg` yet it had to be in `DEPEND`

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
So are you absolutely sure about the header requirement here?

File: media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.24.1401426.ebuild [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+MY_PV_FULL="${MY_PV}-${MY_PV_REV}"
+
+MY_PN="amf-amdgpu-pro"
+
+PRO_VULKAN_PKG_VER="22.10.1"
+
+MY_LINK="https://repo.radeon.com/amdgpu/${PRO_VULKAN_PKG_VER}/ubuntu/pool/proprietary/a/${MY_PN}"
+
+DESCRIPTION="AMD's closed source Advanced Media Framework (AMF) driver"
+HOMEPAGE="https://www.amd.com/en/support"
```
[the remark this answers is not in the document]
Comment (AdelKS) [reply media-video/amdgpu-pro-amf/amdgpu-pro-amf-1.4.24.1401426.ebuild:23]: I cannot find anything specifically related to AMD's closed source AMF driver

File: media-video/ffmpeg/ffmpeg-4.4.2-r1.ebuild [hunk 2]
```diff
@@ -287,6 +287,7 @@ RDEPEND="${RDEPEND}
 "
 
 DEPEND="${RDEPEND}
+	amf? ( media-libs/amf-headers:= )
```
[the remark this answers is not in the document]
Comment (AdelKS) [reply media-video/ffmpeg/ffmpeg-4.4.2-r1.ebuild:290]: Please correct me if I misunderstood something: `media-libs/amf-headers` only installs headers, no shared nor static libraries. And once `ffmpeg` gets built with those headers, they are unneeded at runtime. So I do not need to put the dependency in RDPEND ?

**Comment by AdelKS** [comment]
> So are you absolutely sure about the header requirement here?

I am not sure I understand your question. For `ffmpeg` to be built with `amf` useflag, the headers are absolutely needed. What I am unsure about is the need to rebuild when updating the headers and I only used common sense on that. 

The thing is that the library is always backward-compatible with all previous AMD drivers, but only the newer ones can leverage the new features. And `ffmpeg` dlopens the shared library installed by `amdgpu-pro-amf`. 

I would like your input on this, or anyone's input with experience on such matters.

File: media-video/ffmpeg/ffmpeg-4.4.2-r1.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (juippis) [reply media-video/ffmpeg/ffmpeg-4.4.2-r1.ebuild:290]: `:=` only works in RDEPEND.

**Comment by AdelKS** [comment]
Hello @juippis 

I cancelled my slotting changes for now, this PR is a simple version bump + a fix of deps in ffmpeg. 

I will wait for the answer on https://github.com/GPUOpen-LibrariesAndSDKs/AMF/issues/313 to do further changes. By the look of their repository, that contains a copy of `ffmpeg` that did not get updated with the recent amf-headers 1.4.24, I think it's either
- ffmpeg needs further patches from AMD to pickup the new features
- the features are controlled by `amdgpu-pro-amf` and not `ffmpeg` who's a simple user

File: media-video/ffmpeg/ffmpeg-4.4.2-r1.ebuild [hunk 3]
```diff
@@ -287,6 +287,7 @@ RDEPEND="${RDEPEND}
 "
 
 DEPEND="${RDEPEND}
+	amf? ( media-libs/amf-headers )
```
Comment (juippis) [inline media-video/ffmpeg/ffmpeg-4.4.2-r1.ebuild:290]: Now this on the other hand doesn't require a revbump to ffmpeg :) rest of this PR works so I guess I'll move and rebase this revbump out. Thanks!

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**